overlord/configstate/configcore/services_test.go

```diff
@@ -437,6 +448,81 @@ func (s *servicesSuite) TestConfigureNetworkValid(c *C) {
 	}
 }
 
+func (s *servicesSuite) TestConfigureNetworkValidSocket(c *C) {
+	s.socketEnabled = true
+	defer func() { s.socketEnabled = false }()
+
+	r := configcore.MockNetLookupIP(func(host string) ([]net.IP, error) {
+		switch host {
+		case "host":
+			return []net.IP{
+				net.ParseIP("127.0.0.1"),
+			}, nil
+		case "2001", "2001.net":
+			return []net.IP{
+				net.ParseIP("127.0.0.2"),
+			}, nil
+		case "host1":
+			return []net.IP{
+				net.ParseIP("127.0.0.3"),
+			}, nil
+		case "host2":
+			return []net.IP{
+				net.ParseIP("127.0.0.4"),
+			}, nil
+		}
+		return nil, fmt.Errorf("unknown host: %s", host)
+	})
+	defer r()
+
+	sshListenCfg := filepath.Join(dirs.GlobalRootDir, "/etc/systemd/system/ssh.socket.d/listen.conf")
+
+	for _, tc := range []struct {
+		confStr string
+		sshConf string
+	}{
+		// valid hostnames/IPs
+		{"host", "[Socket]\nFreeBind=true\nListenStream=\nListenStream=127.0.0.1:22\n"},
+		{"10.0.2.2", "[Socket]\nFreeBind=true\nListenStream=\nListenStream=10.0.2.2:22\n"},
+		{"::1", "[Socket]\nFreeBind=true\nListenStream=\nListenStream=[::1]:22\n"},
+		{"[::1]:8022", "[Socket]\nFreeBind=true\nListenStream=\nListenStream=[::1]:8022\n"},
+		{"2001", "[Socket]\nFreeBind=true\nListenStream=\nListenStream=127.0.0.2:22\n"},
+		{"2001.net", "[Socket]\nFreeBind=true\nListenStream=\nListenStream=127.0.0.2:22\n"},
+		{"2001.net:1234", "[Socket]\nFreeBind=true\nListenStream=\nListenStream=127.0.0.2:1234\n"},
+		// port only
+		{":9022", "[Socket]\nFreeBind=true\nListenStream=\nListenStream=9022\n"},
```

Comment:
Should this one has `FreeBind=true`?

Reply:
No not really needed, it is only needed for IP addresses, but i added it in all cases. I'll get this properly designed